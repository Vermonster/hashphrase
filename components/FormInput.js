import React from 'react';
import { Input } from 'native-base';
import PropTypes from 'prop-types';

const FormInput = ({
  name, placeholder, secureTextEntry, onChange,
}) => (
  <Input
    name={name}
    placeholder={placeholder}
    secureTextEntry={secureTextEntry}
    autoCapitalize="none"
    onChange={e => onChange(e)}
  />
);

FormInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
};

export default FormInput;
