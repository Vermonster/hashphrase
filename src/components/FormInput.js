import React from 'react';
import { View, StyleSheet } from 'react-native';
import { IconButton, Text, TextInput } from 'react-native-paper';
import { colors, fontSize } from '../styles/base';

const styles = StyleSheet.create({
  label: {
    alignSelf: 'flex-start',
    fontSize: fontSize.lg,
  },
  textInput: {
    flex: 1,
    backgroundColor: colors.white,
  },
  formInputContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    backgroundColor: colors.white,
    marginTop: 10,
    width: '80%',
  },
  visibilityButton: {
    position: 'absolute',
    alignSelf: 'center',
  },
});

class FormInput extends React.Component {
  isTextHidden = (inputLabel, visibleStatus) => {
    if (inputLabel === 'label') {
      return false;
    } if (inputLabel === 'password' || inputLabel === 'confirmPassword') {
      if (visibleStatus) {
        return false;
      }
    }
    return true;
  }

  render() {
    const {
      placeholderText,
      prompt,
      value,
      handleChange,
      label,
      componentType,
      error,
      handleInputFocus,
      inputRef,
      handlePasswordVisibility,
      visibility,
    } = this.props;
    const visibilityIcon = visibility ? 'visibility' : 'visibility-off';
    const visibleText = this.isTextHidden(componentType, visibility);
    const buttonType = (componentType === 'password')
      ? (
        <IconButton
          icon={visibilityIcon}
          onPress={handlePasswordVisibility}
          accessibilityLabel="hide or show password"
          style={styles.visibilityButton}
          color={colors.secondary}
          testID="hideShowButton"
        />
      ) : null;

    const inputMargin = componentType === 'confirmPassword' ? 0 : 30;

    return (
      <View>
        <Text style={styles.label}>{prompt}</Text>
        <View style={[styles.formInputContainer, { marginBottom: inputMargin }]}>
          <TextInput
            value={value}
            placeholder={placeholderText}
            secureTextEntry={visibleText}
            style={styles.textInput}
            onChange={handleChange(componentType)}
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect={false}
            label={label}
            error={error}
            returnKeyType="next"
            onSubmitEditing={handleInputFocus}
            ref={inputRef}
            disableFullscreenUI
          />
          { buttonType }
        </View>
      </View>
    );
  }
}

export default FormInput;
