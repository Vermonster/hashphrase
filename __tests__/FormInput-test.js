import React from 'react';
import { shallow } from 'enzyme';
import FormInput from '../src/components/FormInput';

describe('<FormInput />', () => {
  const mockFn = jest.fn();

  it('should render the same snapshot', () => {
    const wrapper = shallow(
      <FormInput
        placeholderText="Test Placeholder"
        prompt="Test Prompt"
        handleChange={mockFn}
        className="test-class-name"
        label="password"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the component state when toggleVisibility is called', () => {
    const wrapper = shallow(
      <FormInput
        placeholderText="Test Placeholder"
        prompt="Test Prompt"
        handleChange={mockFn}
        className="test-class-name"
        label="password"
      />,
    );
    expect(wrapper.state('visible')).toBe(false);
    wrapper.instance().toggleVisibility();
    expect(wrapper.state('visible')).toBe(true);
  });

  it('should generate a password input when "password" classname provided', () => {
    const wrapper = shallow(
      <FormInput
        placeholderText="Test Placeholder"
        prompt="Test Prompt"
        handleChange={mockFn}
        className="password"
        label="password"
      />,
    );
    expect(wrapper.find('.submitPassword')).toHaveLength(1);
  });
});
