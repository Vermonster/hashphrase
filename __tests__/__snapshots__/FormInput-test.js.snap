// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> should render the same snapshot 1`] = `
<ForwardRef(View)>
  <withTheme(Text)
    style={
      Object {
        "alignSelf": "flex-start",
        "fontSize": 16,
      }
    }
  >
    Test Prompt
  </withTheme(Text)>
  <ForwardRef(View)
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginTop": 10,
          "width": "80%",
        },
        Object {
          "marginBottom": 30,
        },
      ]
    }
  >
    <withTheme(TextInput)
      autoCapitalize="none"
      autoComplete="off"
      autoCorrect={false}
      placeholder="Test Placeholder"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    />
    <withTheme(IconButton)
      accessibilityLabel="hide or show password"
      color="#212A59"
      icon="visibility-off"
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "position": "absolute",
        }
      }
      testID="hideShowButton"
    />
  </ForwardRef(View)>
</ForwardRef(View)>
`;
